src/metabase/task/sync.clj

```diff
@@ -0,0 +1,152 @@
+  [database]
+  (doseq [{:keys [action]} sfc-job-definitions]
+    (let [[trigger-name job-name] (db-task-names database action)
+          [job-key trigger-key] (db-task-keys trigger-name job-name)]
+      (log/info (u/format-color 'cyan "unscheduling %s for database-id: '%s' named %s"
+                  action
+                  (:name database)
+                  job-name))
+      (task/delete-task! job-key trigger-key))))
+
+
+(defn- sfc-job [db-id job-type job-key]
+  (jobs/build
+   (jobs/of-type job-type)
+   (jobs/using-job-data {"db-id" db-id})
+   (jobs/with-identity job-key)))
+
+(defn- trigger-with-cron-schedule [trigger-key cron-schedule]
+  (triggers/build
+   (triggers/with-identity trigger-key)
+   (triggers/start-now)
+   (triggers/with-schedule
+     (cron/schedule
+      (cron/cron-schedule cron-schedule)
+      ;; drop tasks if they start to back up
+      (cron/with-misfire-handling-instruction-do-nothing)))))
+
+(defn- sfc-job-and-trigger [database {:keys [action schedule-key job-type]}]
+  (let [cron-schedule           (get database schedule-key)
+        [trigger-name job-name] (db-task-names (u/get-id database) action)
+        [job-key trigger-key]   (db-task-keys trigger-name job-name)]
+    {:job      (sfc-job (u/get-id database) job-type job-key)
+     :job-name job-name
+     :trigger  (trigger-with-cron-schedule trigger-key cron-schedule)
+     :schedule cron-schedule}))
+
+(defn schedule-db-sync-actions!
+  "Schedule the Sync, Analyze, Cache-field-values, and classify jobs for a database
+   Deletes and replaces any existing schedules"
+  [database]
+  (unschedule-all-tasks-for-db! database)
+  (doseq [job-def sfc-job-definitions]
+    (let [{:keys [job job-name trigger schedule]} (sfc-job-and-trigger database job-def)]
+      (log/info (u/format-color 'green "scheduling %s for database: '%s' at %s named %s" (:action job-def) (:name database) schedule job-name))
+      ;; submit a new job to the scheduler, but only if it still exists. this function is called when a DB is deleted. (TODO - huh?)
+      (task/schedule-task! job trigger)
+      job-name)))
+
+
+;;; +------------------------------------------------------------------------------------------------------------------------+
+;;; |                                                       TASK INIT                                                        |
+;;; +------------------------------------------------------------------------------------------------------------------------+
+
+(defonce ^:private classify-databases-job (atom nil))
+
+(defn task-init
+  "classify called during startup; start the job for classify databases."
+  []
+  ;; build one job and one trigger for each database.
+  (let [triggers (doseq [database (db/select Database, :is_sample false)]
+                   (schedule-db-sync-actions! database))] ;; we're building a sequence of these jobs so they can be stopped later
+    (reset! classify-databases-job triggers)))
```

Comment:
This PR's priorities:
1. allow for separate schedules of fields getting their values cached vs getting analyzed/fingerprinted
2. Allow for an admin facing UX to set the field value caching schedule + allow manual re-caches of fields + databases
3. Allow value caching to happen via samples. Bonus points for allowing a "do a tablescan" override on this.